### PR TITLE
Homework paskar oleksandr pure jdbc

### PR DESCRIPTION
**Pull request by Paskar-Oleksandr**

### CODE REVIEW

File: src/main/java/com/paskar/email/application/service/EmailRepositoryJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@AllArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class EmailRepositoryJdbcImpl implements EmailRepository {
+
+    JdbcTemplate jdbcTemplate;
```
Comment (vova-pereb) [inline src/main/java/com/paskar/email/application/service/EmailRepositoryJdbcImpl.java:22]: Not part of JDBC task.

File: src/main/java/com/paskar/email/application/service/EmailRepositoryPureJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        } catch (SQLException exception) {
+            exception.printStackTrace();
+        }
+        return email;
+    }
+
+    @Override
+    public List<Email> findAll() {
+        List<Email> list = new ArrayList<>();
+        String sql = "SELECT * FROM first_homework_with_jdbc";
```
Comment (vova-pereb) [inline src/main/java/com/paskar/email/application/service/EmailRepositoryPureJdbcImpl.java:81]: Avoid SELECT *

File: src/main/java/com/paskar/email/application/service/EmailRepositoryPureJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            preparedStatement.executeUpdate();
+
+        } catch (SQLException exception) {
+            exception.printStackTrace();
+        }
+    }
+
+    @Override
+    public List<Email> findEmailsNearDeliveryDate() throws IOException {
+        return null;
```
Comment (vova-pereb) [inline src/main/java/com/paskar/email/application/service/EmailRepositoryPureJdbcImpl.java:146]: It is better to return Collections#emptyList()

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+spring.mail.properties.mail.smtp.auth=true
+spring.mail.properties.mail.smtp.connectiontimeout=5000
+spring.mail.properties.mail.smtp.timeout=5000
+spring.mail.properties.mail.smtp.writetimeout=5000
+spring.mail.properties.mail.smtp.starttls.enable=true
+
+spring.jackson.serialization.write_dates_as_timestamps=false
+
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.url=jdbc:mysql://${MYSQL_HOST:localhost}:3306/academy_jdbc_homework
```
Comment (vova-pereb) [inline src/main/resources/application.properties:15]: Not part of JDBC task

**Review by vova-pereb — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 4 above)
Comment (Paskar-Oleksandr) [reply src/main/resources/application.properties:15]: if I delete this configuration from **application.properties** I cannot run the application
What should I do?

Comment (vova-pereb) [reply src/main/resources/application.properties:15]: It means that you made application in wrong way.
You should not use JdbcTemplate / DataSource in this task.

**Pull request closed without merging**